From 9aad9f29295fdafaa6c15ed61f61070e46b8051f Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Wed, 10 Apr 2019 20:16:27 +0200
Subject: [PATCH] TRITON-1332 Include datacenter name in networking.json
 Reviewed by: Jason King <jason.king@joyent.com> Approved by: Chris Burroughs
 <chris.burroughs@joyent.com>

---
 lib/boot-files.js             | 6 +++++-
 lib/net-file.js               | 9 +++++++--
 sapi_manifests/dhcpd/template | 1 +
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/lib/boot-files.js b/lib/boot-files.js
index 5c94d9b..0ae7f5c 100644
--- a/lib/boot-files.js
+++ b/lib/boot-files.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -60,6 +60,10 @@ function extractBootOpts(opts) {
         bootOpts.dnsDomain = opts.config.dnsDomain;
     }
 
+    if (opts.config.datacenterName) {
+        bootOpts.datacenterName = opts.config.datacenterName;
+    }
+
     return bootOpts;
 }
 
diff --git a/lib/net-file.js b/lib/net-file.js
index 6dad0fd..6e18cb0 100644
--- a/lib/net-file.js
+++ b/lib/net-file.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -70,7 +70,8 @@ function generateNetConfFile(opts) {
         nics: opts.nics || [],
         nictags: opts.nictags,
         opts_hostname: opts.hostname || '<undefined>',
-        overlay: opts.overlay || '<undefined>'
+        overlay: opts.overlay || '<undefined>',
+        datacenter_name: opts.datacenterName || '<undefined>'
     }, 'Generating net conf file');
 
     var conf = {
@@ -92,6 +93,10 @@ function generateNetConfFile(opts) {
         conf.hostname = opts.hostname;
     }
 
+    if (opts.datacenterName) {
+        conf.datacenter_name = opts.datacenterName;
+    }
+
     // The admin nic is special in a few ways:
     // - It needs to go first in the nics list (in case there are other
     //   vnics over the admin nic tag)
diff --git a/sapi_manifests/dhcpd/template b/sapi_manifests/dhcpd/template
index 22dbb3b..3e7cb92 100644
--- a/sapi_manifests/dhcpd/template
+++ b/sapi_manifests/dhcpd/template
@@ -21,6 +21,7 @@
   },
   "disableBootTimeFiles": true,
 {{/fabric_cfg}}
+  "datacenterName": "{{datacenter_name}}",
   "dnsDomain": "{{dns_domain}}",{{#http_pxe_boot}}
   "ipxeHTTP": {{http_pxe_boot}},{{/http_pxe_boot}}{{#chainload_grub}}
   "chainloadGrub": {{chainload_grub}},{{/chainload_grub}}
-- 
2.21.0

